plankton/core/utils.py

```diff
@@ -115,3 +120,85 @@ def seconds_since(start):
     :return: Elapsed seconds since start time.
     """
     return (datetime.now() - start).total_seconds()
+
+
+class FromOptionalDependency(object):
+    def __init__(self, module, exception=None):
+        """
+        This is a utility class for importing classes from a module or
+        replacing them with dummy types if the module can not be loaded.
+
+        Assume module 'a' that does:
+
+            from b import C, D
+
+        and module 'e' which does:
+
+            from a import F
+
+        where 'b' is a hard to install dependency which is thus optional.
+        To still be able to do:
+
+            import e
+
+        without raising an error, for example for inspection purposes,
+        this class can be used as a workaround in module 'a':
+
+            C, D = FromOptionalDependency('b').do_import('C', 'D')
+
+        which is not as pretty as the actual syntax, but at least it
+        can be read in a similar way. If the module 'b' can not be imported,
+        stub-types are created that are called 'C' and 'D'. Everything depending
+        on these types will work until any of those are instantiated - in that
+        case an exception is raised.
+
+        The exception can be controlled via the exception-parameter. If it is a
+        string, a PlanktonException is constructed from it. Alternatively it can
+        be an instance of an exception-type. If not provided, a PlanktonException
+        with a standard message is constructed. If it is anything else, a RuntimeError
+        is raised.
+
+        Essentially, this class helps deferring ImportErrors until anything from
+        the module that was attempted to load is actually used.
+
+        :param module: Module from that symbols should be imported.
+        :param exception: Text for PlanktonException or custom exception object.
+        """
+        self._module = module
+
+        if exception is None:
+            exception = 'The optional dependency \'{}\' is required for the ' \
+                        'functionality you tried to use.'.format(self._module)
+
+        if isinstance(exception, string_types):
+            exception = PlanktonException(exception)
+
+        if exception is not None and not isinstance(exception, BaseException):
+            raise RuntimeError(
+                'The exception parameter has to be either a string or a an instance of an '
+                'exception type (derived from BaseException).')
```

Comment:
True, exception could be `None` in an earlier version of this, but now it can't. Fixing it. But I think it should be a `RuntimeError`, because giving anything else than an exception type or a string is a programming error and it is only something to notify developers, not users.